### PR TITLE
Private forks should not be deletable by upstream

### PR DESCRIPTION
**Pull request by phtan**
This pull request addresses issue #2863

### CODE REVIEW

File: models/repo.go [hunk 1]
```diff
@@ -1409,21 +1409,9 @@ func DeleteRepository(uid, repoID int64) error {
 	}
 
 	if repo.NumForks > 0 {
-		if repo.IsPrivate {
```
Comment (Unknwon) [inline models/repo.go]: You would need to set `is_fork` to `false`...

**Comment by phtan** [comment]
This pull request addresses issue #3231 too

File: conf/locale/locale_en-US.ini [hunk 2]
```diff
@@ -604,9 +604,9 @@ settings.delete = Delete This Repository
 settings.delete_desc = Once you delete a repository, there is no going back. Please be certain.
 settings.delete_notices_1 = - This operation <strong>CANNOT</strong> be undone.
 settings.delete_notices_2 = - This operation will permanently delete the everything of this repository, including Git data, issues, comments and accesses of collaborators.
-settings.delete_notices_fork_1 = - If this repository is public, all forks will become independent after deletion.
-settings.delete_notices_fork_2 = - If this repository is private, all forks will be removed at the same time.
-settings.delete_notices_fork_3 = - If you want to keep all forks after deletion, please change visibility of this repository to public first.
+settings.delete_notices_fork_1 = - All forks will become independent after deletion.
+settings.delete_notices_fork_2 =
```
Comment (Unknwon) [inline conf/locale/locale_en-US.ini]: These two lines could be removed as well.

**Comment by phtan** [comment]
@Unknwon I've removed the lines that you mentioned.

**Comment by Unknwon** [comment]
Thanks, please resolve the conflict and clean up to 1 commit, should be good then.

**Comment by phtan** [comment]
Done, @Unknwon

**Comment by Unknwon** [comment]
Thanks!

**Comment by phtan** [comment]
Thank you @Unknwon for your effort on this little PR (yes, it has been rather close to two years since)

**Pull request merged**